addons/mail/static/src/components/discuss/discuss.js

```diff
@@ -25,17 +25,19 @@ class Discuss extends Component {
     constructor(...args) {
         super(...args);
         useStore(props => {
-            const discuss = this.env.messaging.discuss;
+            const discuss = this.env.messaging && this.env.messaging.discuss;
             return {
-                checkedMessages: discuss.threadViewer.checkedMessages.map(message => message.__state),
-                discuss: discuss.__state,
-                isDeviceMobile: this.env.messaging.device.isMobile,
-                isMessagingInitialized: this.env.messaging.isInitialized,
-                thread: discuss.thread ? discuss.thread.__state : undefined,
-                threadCache: discuss.threadViewer.threadCache
+                checkedMessages: discuss ? discuss.threadViewer.checkedMessages.map(message => message.__state) : [],
+                discuss: discuss ? discuss.__state : undefined,
+                isDeviceMobile: this.env.messaging && this.env.messaging.device.isMobile,
+                isMessagingInitialized: this.env.isMessagingInitialized(),
+                thread: discuss && discuss.thread ? discuss.thread.__state : undefined,
+                threadCache: (discuss && discuss.threadViewer && discuss.threadViewer.threadCache)
                     ? discuss.threadViewer.threadCache.__state
                     : undefined,
-                uncheckedMessages: discuss.threadViewer.uncheckedMessages.map(message => message.__state),
+                uncheckedMessages: discuss && discuss.threadViewer
+                    ? discuss.threadViewer.uncheckedMessages.map(message => message.__state)
+                    : [],
```

Comment:
changes in this file are overkill with the fix on the widget, but ok to keep them just because I think the widget should ideally allow a spinner.

Reply:
I would keep fix as simple as possible for discuss widget, therefore keeping the `await` and not have spinner.
I would also keep changes in components, because the intend is to remove widget eventually so I think it's good not making too loose assumptions from widget like here.

Reply:
ok to keep